### PR TITLE
Fixed bug in add() pipe validation

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #22 . 
Changes in methods: `TkPipeAdd.add(String)` and `DyPipes.add(String)`. Other lines in some files ($Id$) were automatically changed by git when I checked them out. 

Also, I fixed a mistake in .gitattribtues . Specified `eol=lf` for `.java` and `.xml` files specifically, since it breaks the images if I simply specify `*`

### CODE REVIEW

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by amihaiemil** [comment]
@yegor256  git acts funny about that $Id$ . I have the same behaviour when I checkout the repo and try to switch to branch 22. Did you ever encounter this before?

**Comment by yegor256** [comment]
@amihaiemil I think you should not touch that `$Id$` placeholders in source files. they are replaced by Git automatically

**Comment by amihaiemil** [comment]
@yegor256 I reverted the changes to the ``$Id$``. Something seems off anyway - git automatically removes the id when I checkout the files and they all appear modified in my local repo. 

Plus, if you look in the history apparently the files had the same id when I first checked them out
``c79829f9e91907f21c716854779af4233e496fa9``

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@amihaiemil I think you need to merge master into your branch

**Comment by amihaiemil** [comment]
@yegor256  try now, maybe? I had to do the merge manually since git wouldn't let me. It changes the ID instantly, I can't even do ``git checkout -- .`` : the operation works fine, but then I do ``git status`` and changes are there again :))

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by yegor256** [comment]
@amihaiemil what's up with this PR?

**Comment by amihaiemil** [comment]
@yegor256 it's so old I don't even remember. It's probably not even applicable anymore. Closing it :)

**Comment by yegor256** [comment]
@amihaiemil it's still applicable, the bug is still there. It's a pity you closed it, the code was correct, there were just merge problems.

**Comment by 0crat** [comment]
@yegor256 please, pay attention to this pull request

**Comment by 0crat** [comment]
Job `gh:yegor256/wring#26` is in scope.

**Comment by amihaiemil** [comment]
@yegor256 reopened :) I'll have a look next week

**Comment by yegor256** [comment]
@amihaiemil thanks

File: src/main/java/io/wring/tk/TkPipeAdd.java [hunk 1]
```diff
@@ -62,13 +66,27 @@
...
+                Json.createReader(
+                    IOUtils.toInputStream(
+                        new RqForm.Base(req).param("json").iterator().next()
+                            .trim()
+                    )
+                ).readObject().toString()
+            );
+            message = "pipe created";
+        } catch (final JsonParsingException ex) {
+            message = "Invalid json format. No pipe added.";
```
Comment (yegor256) [inline src/main/java/io/wring/tk/TkPipeAdd.java]: @amihaiemil we won't show enough information to the user. The exception text will be lost. Let's add `ex.getMessage()` to the message.

**Comment by yegor256** [comment]
@amihaiemil I'm sorry it took two years to review your PR :) Please, fix the code above and let's merge it, it's a good PR.

**Comment by amihaiemil** [comment]
@yegor256 I have no idea what's happening, some settings may be not ok in this repo I cloned my fork locally and it had about 10 files changed automatically, that git $Id$ deleted... I remember I had this issue here also 2 years ago

**Comment by yegor256** [comment]
@amihaiemil the bug is still there, I believe. If you post an empty pipe, the system crashes with an exception.

**Comment by amihaiemil** [comment]
@yegor256 I'll try to make another clean PR. Reclone the repo and everything. But I still don't undertand why the weird behaviour with the ``$Id`` -- when I clone it locall git automatically deletes the generated IDs and shows all files as modified. Sure there isn't some misplaced setting somewhere? Never had it in any other repo.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually or invite more people to the project, as explained in [§51](http://www.zerocracy.com/policy.html#51)

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually or invite more people to the project, as explained in [§51](http://www.zerocracy.com/policy.html#51)

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually or invite more people to the project, as explained in [§51](http://www.zerocracy.com/policy.html#51)

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by yegor256** [comment]
@0crat assign me

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/wring/issues/26#issuecomment-386854423))

@yegor256 The job [#26](https://github.com/yegor256/wring/issues/26) assigned to @yegor256[/z](https://www.0crat.com/u/yegor256), here is [why](https://www.0crat.com/footprint/C7FCB1EQN/1260544417); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at this job; I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by yegor256** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/yegor256/wring/issues/26#issuecomment-388818145))

@yegor256 GitHub issue is still open, won't close

**Comment by yegor256** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/yegor256/wring/issues/26#issuecomment-388820632))

@yegor256 This is what I know about this job, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#26](https://github.com/yegor256/wring/issues/26) is [in scope](http://www.zerocracy.com/policy.html#14) for 8days
  * The role is `DEV`
  * The job is assigned to @yegor256[/z](https://www.0crat.com/u/yegor256) for [8days](http://www.zerocracy.com/policy.html#8)
  * There is no monetary reward attached, it's a [free](http://www.zerocracy.com/policy.html#2) job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 30 minutes/points
  * These users are banned and won't be assigned:
    * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/C7FCB1EQN?q=%7Bjob%3A%27gh%3Ayegor256%2Fwring%2326%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` are banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#26](https://github.com/yegor256/wring/issues/26) is now out of scope

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `DEV` is banned at [#26](https://github.com/yegor256/wring/issues/26); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
The job [#26](https://github.com/yegor256/wring/issues/26) is now out of scope

**Pull request closed without merging**